### ISSUE TITLE
appium color in logs

### ISSUE THREAD

**Issue by kazzmir** [appium/appium#12021]
## The problem

Redirecting the output of appium to a logfile ends up with xterm color codes. This behavior is new in appium 1.10.1.

## Environment

* Appium version (or git revision) that exhibits the issue: 1.10.1
* Last Appium version that did not exhibit the issue (if applicable): 1.9.1
* Desktop OS/version used to run Appium: linux

## Reproduce

Run appium *without* the `--log-no-colors` option while redirecting the output to a logfile. It works best when run by another process that can control the environment.
```
$ cat x.py
import subprocess
with open("x", "w") as f:
    subprocess.call(['node_modules/.bin/appium'], env={}, stdout=f)
$ python x.py
```
Then look at the output 'x' (you might have to ctrl-c the script). On appium 1.9.1 there are no xterm color codes, but in 1.10.1 there are

1.9.1
```
$ xxd x
00000000: 5b41 7070 6975 6d5d 2057 656c 636f 6d65  [Appium] Welcome
00000010: 2074 6f20 4170 7069 756d 2076 312e 392e   to Appium v1.9.
00000020: 310a                                     1.
```

1.10.`
```
$ xxd x
00000000: 1b5b 3335 6d5b 4170 7069 756d 5d1b 5b33  .[35m[Appium].[3
00000010: 396d 2057 656c 636f 6d65 2074 6f20 4170  9m Welcome to Ap
00000020: 7069 756d 2076 312e 3130 2e31 0a         pium v1.10.1.
```

It looks like the `winston` module that appium uses for logging has changed in that it no longer looks at the `TERM` environment variable to detect if color is supported. Even this code is in a dependency of winston, called `colors`, so perhaps this bug report should be made against `colors` or `winston` rather than `appium`.

1.9.1:
```
node_modules/appium/node_modules/winston/node_modules/colors/lib/system/supports-colors.js
52:  if (process.env.TERM === 'dumb') {
56:  if (/^screen|^xterm|^vt100|color|ansi|cygwin|linux/i.test(process.env.TERM)) {
```

In any case it suffices to pass `--log-no-colors`, but just wanted to make you aware of the change in behavior.

**Comment by mykola-mokhnach** [comment]
@imurchie FYI

**Comment by imurchie** [comment]
This has been fixed (#11853) and will be in the next release.
